### PR TITLE
Persisting auth token and app locale

### LINKED ISSUE #146: Persist auth token
## Description
The auth token is not being persisted anywhere in the client, so if the user refreshes the page the authentication is lost, so we need to persist that token and allow the user to reuse it for the lifespan of the token even if they refresh the page.

## Acceptance Criteria
- Auth token should persist, either via cookies or session storage

User story: N/A\
Pre-requisites: N/A

### PR DESCRIPTION
**Pull request by gilhrpenner**
# What I did
Created vuex store for the app to persist the following:
- Authentication status
- Auth token
- App language

Closes #146  #148

### CODE REVIEW

File: frontend/src/pages/auth/Login.vue [hunk 1]
```diff
@@ -111,19 +112,23 @@ const onLoginSubmit = () => {
...
 
-            // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
-            api.defaults.headers.common['Authorization'] = `Bearer ${bearer}`;
-
             store.commit('account/setAccount', account);
+            store.commit('app/setAuthToken', bearer);
+
             void router.push('/dashboard');
         }).catch((error: AxiosRejectResponse): void => {
+            console.log(error);
```
Comment (akashd50) [inline frontend/src/pages/auth/Login.vue:120]: Do we wanna remove this or keep it for debugging?

Comment (MarkRobitaille) [reply frontend/src/pages/auth/Login.vue:120]: I was curious about this too

Comment (gilhrpenner) [reply frontend/src/pages/auth/Login.vue:120]: Good catch, forgot to handle that properly.

Comment (gilhrpenner) [reply frontend/src/pages/auth/Login.vue]: Done

**Pull request merged**